WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewController.swift

```diff
@@ -69,6 +77,8 @@ final class BlogDashboardViewController: UIViewController {
         super.viewDidAppear(animated)
 
         viewModel.loadCards()
+        DashboardPostsSyncManager.shared.syncPosts(blog: blog, statuses: draftStatusesToSync)
+        DashboardPostsSyncManager.shared.syncPosts(blog: blog, statuses: scheduledStatusesToSync)
```

Comment:
What do you think about moving that to `BlogDashboardService`? I like that for a few reasons:

1. You can check if there are any draft or scheduled posts being returned and sync those only if necessary
2. This is tied to `loadCards` which in my opinion makes sense. As it is right now, if for any reason we decide to stop returning the draft/scheduled cards, the app will still be loading the posts, even if it doesn't have to (think about the future in which we would only show a pages card, we will have to change this anyway)

I think making this logic dependent on the returned cards makes sense.

Ps.: again, ideally, the API should return the posts. But that wasn't possible. :(

Reply:
+1 to moving `syncPosts` to `BlogDashboardService`

Reply:
@leandroalonso 
> I think making this logic dependent on the returned cards makes sense.

I agree! However, I thought it's better to move the `syncPosts` logic to `BlogDashboardViewModel` instead of `BlogDashboardService`. This avoided some duplication between remote fetch and local fetch paths in `BlogDashboardService`.

Changed in 6fc1dfa807fcb09c665715c3ff66d038926c04fd
Let me know what you think!